utils/manifest.ts

```diff
@@ -135,14 +135,35 @@ export const createFunctionInstanceFromFunctionKey = (
   return functionInstance;
 };
 
+
+/**
+ * The database may have more functions than what's referenced by the sections.
+ * Maybe, this is due to a bug on the live editor's interface. Anyways, it's always
+ * good to prune and only run the necessary functions once
+ */
+const pruneFunctions = (data: PageData) => {
+  const { sections, functions } = data;
+  const str = JSON.stringify(sections);
+
+  const functionsMap = new Map<string, PageData["functions"][number]>();
+
+  for (const fn of functions) {
+    if (str.includes(fn.uniqueId)) {
+      functionsMap.set(fn.uniqueId, fn);
+    }
+  }
+
+  return [...functionsMap.values()];
+};
+
 export async function loadPageData<Data, State extends LiveState>(
   req: Request,
   ctx: HandlerContext<Data, State>,
   pageData: PageData,
 ): Promise<PageData> {
   const { start, end } = ctx.state.t;
   const functionsResponse = await Promise.all(
-    pageData.functions?.map(async ({ key, props, uniqueId }) => {
+    pruneFunctions(pageData).map(async ({ key, props, uniqueId }) => {
```

Comment:
Wdyt about prunning sections?

Reply:
We don't need to prune sections because the user already prunes the sections on the editor by adding/removing sections. We have to prune functions because users don't prune functions visually